fedbiomed/common/training_plans/_base_training_plan.py

```diff
@@ -86,6 +85,11 @@ def post_init(
             aggregator_args: Arguments managed by and shared with the
                 researcher-side aggregator.
         """
+        # Set random seed
+        rseed = training_args['random_seed']
+        if rseed is not None:
+            random.seed(rseed)
```

Comment:
Hi @sharkovsky,

As I understand random.seed for mainly for; 
- model initialization 
- data shuffling 

However, model initialization is never done on the node side. Node always gets initial parameters from the researcher. So since it is possible for researcher set random seed in his python runtime i consider that it is more useful for training data. 

However, random seed is set in `post_init` so it will impact every method that uses randomness.  My concern is that how it will  impact other parts of training for example DP. 

-

Reply:
My purpose with `random_seed` was to maximize reproducibility. That is why I tried to set the random seed as soon as possible and with a global method. I am happy if it also affects DP. 

Maybe I can make this clearer in the documentation, since I only spoke about model and data there. Really, the purpose is to enable:
- reproducible results
- debugging